record/internal.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package record
+
+import (
+	"errors"
+	"hash"
+	"hash/crc32"
+	"math"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/tsdb/chunks"
+	"github.com/prometheus/tsdb/fileutil"
+	"github.com/prometheus/tsdb/labels"
+)
+
+var (
+	// ErrOutOfOrderSample is returned if an appended sample has a
+	// timestamp smaller than the most recent sample.
+	ErrOutOfOrderSample = errors.New("out of order sample")
```

Comment:
These error messages do not seem related to records, but more appenders/series. Sometimes ErrNotFound is used for missing files. Is there a better place for them?

Reply:
`ErrNotFound` could live in `fileutil` I think. `ErrOutOfOrderSample` sample is only used by `MemSeries`. Maybe the package `record` should just have a different name?

Reply:
A different name could help, not totally sure what the best name is though.

Reply:
I think `ErrOutOfOrderSample` and `ErrAmendSample` can go back into the base package now correct since they only relate to appenders?

I am still not sure what to do with `ErrNotFound`, it seems to be used both for files not being found and for series or chunks not being found. Neither of which are related to records.